addons/web/controllers/home.py

```diff
@@ -171,4 +171,19 @@ def health(self, db_server_status=False):
 
     @http.route(['/robots.txt'], type='http', auth="none")
     def robots(self, **kwargs):
-        return "User-agent: *\nDisallow: /\n"
+        allowed_routes = self._get_allowed_robots_routes()
+        robots_content = ["User-agent: *", "Disallow: /"]
+
+        if allowed_routes:
+            for route in allowed_routes:
+                robots_content.append(f"Allow: {route}")
```

Comment:
Mostly for conciseness but fyi it's always better to extend if you're appending some known-size sequence (notably avoids resizing/relocating the underlying array multiple times)
```suggestion
        robots_content.extend(f"Allow: {route}" for route in allowed_routes)
```